src/test/java/com/wiley/autotest/framework/pages/TestElementPage.java

```diff
@@ -81,4 +79,9 @@ public TestElementPage checkNonExistingElementClickFails() {
         element(By.cssSelector("not_present_element"), new SearchStrategy(1)).click();
         return this;
     }
+
+    public TestElementPage checkElemebtHasAnyText() {
+        element(By.xpath("//*[@id=\"men\"]/tbody/tr[1]/td[1]")).should().haveText();
```

Comment:
I don't see how do you check that there is text in the element you point to? What if the text is "". You will need to add some additional assertion here to assert value of this element's text.

Reply:
you also need to check that your method will fail in case the text is null or empty